roottest/root/io/json/CMakeLists.txt

```diff
@@ -1,3 +1,58 @@
 
-ROOTTEST_COMPILE_MACRO(test_classes.h FIXTURES_SETUP json-test-classes-build)
-ROOTTEST_ADD_AUTOMACROS(FIXTURES_REQUIRED json-test-classes-build)
+ROOTTEST_ADD_TEST(ArrayCompress
+                  MACRO runArrayCompress.C
+                  OUTREF ArrayCompress.ref)
+
+ROOTTEST_ADD_TEST(PolyMarker
+                  MACRO runPolyMarker.C
+                  OUTREF PolyMarker.ref)
+
+ROOTTEST_GENERATE_DICTIONARY(libJsonTestClasses
+                  test_classes.h
+                  LINKDEF LinkDef.h
+                  NO_ROOTMAP NO_CXXMODULE)
+
+ROOTTEST_ADD_TEST(BasicTypes
+                  MACRO runBasicTypes.C
+                  OUTREF BasicTypes.ref
+                  DEPENDS ${GENERATE_DICTIONARY_TEST})
```

Comment:
Rather than `depends` we are migrating to using `FIXTURES_SETUP/REQUIRED` which have the added benefit of running the `SETUP` even if you only request to run the actual test.